contentgrid-spring-boot-autoconfigure/src/main/java/com/contentgrid/spring/boot/autoconfigure/security/MultiTenantOAuth2ResourceServerAutoConfiguration.java

```diff
@@ -0,0 +1,79 @@
+package com.contentgrid.spring.boot.autoconfigure.security;
+
+import jakarta.servlet.http.HttpServletRequest;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication.Type;
+import org.springframework.boot.autoconfigure.security.oauth2.resource.servlet.OAuth2ResourceServerAutoConfiguration;
+import org.springframework.boot.autoconfigure.security.servlet.SecurityAutoConfiguration;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.security.authentication.AuthenticationManagerResolver;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.oauth2.server.resource.authentication.JwtIssuerAuthenticationManagerResolver;
+import org.springframework.security.web.SecurityFilterChain;
+
+@AutoConfiguration(before = { OAuth2ResourceServerAutoConfiguration.class, SecurityAutoConfiguration.class })
+@EnableConfigurationProperties(MultiTenantOAuth2Properties.class)
+@ConditionalOnClass(value = {JwtIssuerAuthenticationManagerResolver.class, SecurityFilterChain.class, HttpSecurity.class})
+@ConditionalOnWebApplication(type = Type.SERVLET)
+class MultiTenantOAuth2ResourceServerAutoConfiguration {
+
+
+    @Bean
+    @ConditionalOnMissingBean
+    @ConditionalOnProperty("contentgrid.security.oauth2.trusted-jwt-issuers[0]")
```

Comment:
```suggestion
    @ConditionalOnProperty("contentgrid.security.oauth2.trusted-jwt-issuers")
```

Doesn't this work the same way?

Or do we only want to enable this when there is at least one trusted issuer?

Reply:
Also, shouldn't this whole class be conditional on that property, given that `jwtIssuerSecurityFilterChain` is dependent on this bean being created?